### ISSUE TITLE
AssertionError in Parser.parse() when switch case statement missing trailing semi-colon

### ISSUE THREAD

**Issue by archiecobbs** [mozilla/rhino#796]
Got an `AssertionError` in rhino 1.7.6 in `TokenStream.getLine()` during a unit test (during which assertions are enabled).

To reproduce, first compile this class:

```java
import org.mozilla.javascript.*;
import org.mozilla.javascript.ast.*;

public class xx {

    public static void main(String[] args) throws Exception {
        final String source = "switch (x) { default: throw y }";
        final CompilerEnvirons compilerEnvirons = CompilerEnvirons.ideEnvirons();
        final ErrorCollector errorCollector = new ErrorCollector();
        compilerEnvirons.setErrorReporter(errorCollector);
        final Parser parser = new Parser(compilerEnvirons, errorCollector);
        parser.parse(source, "script", 1);
    }
}
```

```shell
$ javac -classpath .:/Users/archie/.m2/repository/org/mozilla/rhino/1.7.6/rhino-1.7.6.jar
```

Then run using Rhino 1.7.6 with assertions enabled:

```shell
$ java -ea  -classpath .:/Users/archie/.m2/repository/org/mozilla/rhino/1.7.6/rhino-1.7.6.jar xx
Exception in thread "main" java.lang.AssertionError
	at org.mozilla.javascript.TokenStream.getLine(TokenStream.java:1461)
	at org.mozilla.javascript.Parser.warnMissingSemi(Parser.java:3547)
	at org.mozilla.javascript.Parser.autoInsertSemicolon(Parser.java:1095)
	at org.mozilla.javascript.Parser.statementHelper(Parser.java:1077)
	at org.mozilla.javascript.Parser.statement(Parser.java:934)
	at org.mozilla.javascript.Parser.switchStatement(Parser.java:1194)
	at org.mozilla.javascript.Parser.statementHelper(Parser.java:983)
	at org.mozilla.javascript.Parser.statement(Parser.java:934)
	at org.mozilla.javascript.Parser.parse(Parser.java:573)
	at org.mozilla.javascript.Parser.parse(Parser.java:511)
	at xx.main(xx.java:12)
```

**Comment by gbrail** [comment]
This seems to have been fixed since 1.7.6 -- on the current master branch it works fine for me.

**Comment by p-bakker** [comment]
Closing as it cannot be reproduced on the latest master

Please reopen if still reproducible with 1.7.13
